src/Features/CSharp/Portable/CodeRefactorings/NodeSelectionHelpers.cs

```diff
@@ -4,22 +4,63 @@
 
 #nullable disable
 
+using System;
+using System.Collections.Immutable;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings
 {
     internal static class NodeSelectionHelpers
     {
-        internal static async Task<SyntaxNode> GetSelectedDeclarationOrVariableAsync(CodeRefactoringContext context)
+        internal static async Task<ImmutableArray<SyntaxNode>> GetSelectedDeclarationsOrVariablesAsync(CodeRefactoringContext context)
         {
-            // Consider:
-            // MemberDeclaration: member that can be declared in type (those are the ones we can pull up) 
-            // VariableDeclaratorSyntax: for fields the MemberDeclaration can actually represent multiple declarations, e.g. `int a = 0, b = 1;`.
-            // ..Since the user might want to select & pull up only one of them (e.g. `int a = 0, [|b = 1|];` we also look for closest VariableDeclaratorSyntax.
-            return await context.TryGetRelevantNodeAsync<MemberDeclarationSyntax>().ConfigureAwait(false) as SyntaxNode ??
-                await context.TryGetRelevantNodeAsync<VariableDeclaratorSyntax>().ConfigureAwait(false);
+            var (doc, span, cancellationToken) = context;
```

Comment:
```suggestion
            var (document, span, cancellationToken) = context;
```

In general we do not abbreviate unless it's a super common case (like Xml) or ht e name is just excessively long.  For a simpel name like `document` it keeps things nicely consistent to just use that name everywhere.